src/ragas/metrics/_summarization_metric.py

```diff
@@ -0,0 +1,412 @@
+                "Criticism and Challenges": "It faces criticism for risk management and legal issues."
+            }
+        },
+        {
+            "summary": """Photosynthesis is the process by which green plants and certain organisms convert light energy into chemical energy, producing oxygen and organic compounds from water, carbon dioxide, and minerals. It is crucial for life on Earth, as its absence would result in a lack of food and oxygen, causing most organisms to disappear. Only chemosynthetic bacteria, which use chemical energy from inorganic compounds, would survive. Photosynthesis from millions of years ago created fossil fuels, essential for modern society, but their rapid consumption is increasing carbon dioxide levels at an unprecedented rate, significantly impacting Earth's climate.""",
+            "topics": [
+                "Photosynthesis process",
+                "Importance of photosynthesis for life on Earth",
+                "Impact of photosynthesis on atmospheric oxygen",
+                "Chemosynthetic bacteria and their role",
+                "Energy production from photosynthesis and fossil fuels",
+                "Formation of fossil fuels and their significance",
+                "Environmental implications of rapid carbon dioxide rise",
+            ],
+            "summary_topics": {
+                "Photosynthesis process": "Photosynthesis is the process by which green plants and certain organisms convert light energy into chemical energy, producing oxygen and organic compounds from water, carbon dioxide, and minerals.",
+                "Importance of photosynthesis for life on Earth": "It is crucial for life on Earth, as its absence would result in a lack of food and oxygen, causing most organisms to disappear.",
+                "Impact of photosynthesis on atmospheric oxygen": "It is crucial for life on Earth, as its absence would result in a lack of food and oxygen, causing most organisms to disappear.",
+                "Chemosynthetic bacteria and their role": "Only chemosynthetic bacteria, which use chemical energy from inorganic compounds, would survive.",
+                "Energy production from photosynthesis and fossil fuels": "Photosynthesis from millions of years ago created fossil fuels, essential for modern society.",
+                "Formation of fossil fuels and their significance": "Photosynthesis from millions of years ago created fossil fuels, essential for modern society.",
+                "Environmental implications of rapid carbon dioxide rise": "Their rapid consumption is increasing carbon dioxide levels at an unprecedented rate, significantly impacting Earth's climate."
+            }
+        }
+    ]
+)
+
+GENERATE_QUESTION_INSTRUCTION = """\
```

Comment:
The entity/keyword extraction should be separate tasks and not part of the same prompt that creates the question. I would create two prompts
1) To extract entities and or keyphrases from given text
2) Generate questions from text given text and keyphrases and or entities

Although mering this to one improves efficiency, it lacks robustness.

Reply:
@shahules786 Done, made the changes. 
Though I have a concern that how would we regulate the number of questions generated for a text?